e2e/linter/src/linter.test.ts

```diff
@@ -435,6 +436,75 @@ describe('Linter', () => {
         );
       });
     });
+
+    describe('dependency checks', () => {
+      beforeAll(() => {
+        updateJson('.eslintrc.json', (json) => {
+          json.overrides = [
+            ...json.overrides,
+            {
+              files: ['*.json'],
+              parser: 'jsonc-eslint-parser',
+              rules: {
+                '@nx/dependency-checks': 'error',
+              },
+            },
+          ];
+          return json;
+        });
+        updateJson(`libs/${mylib}/project.json`, (json) => {
+          json.targets.lint.options.lintFilePatterns = [
+            `libs/${mylib}/**/*.ts`,
+            `libs/${mylib}/project.json`,
+            `libs/${mylib}/package.json`,
+          ];
+          return json;
+        });
+      });
+
+      it('should report dependency check issues', () => {
+        const nxVersion = readJson('package.json').devDependencies.nx;
+
+        let out = runCLI(`lint ${mylib}`, { silenceError: true });
+        expect(out).toContain('All files pass linting');
+
+        // make an explict dependency to nx
+        updateFile(
+          `libs/${mylib}/src/lib/${mylib}.ts`,
+          (content) =>
+            `import { nxVersion } from 'nx/src/utils/versions';\n\n` +
+            content.replace(/return .*;/, `return nxVersion;`)
+        );
+
+        // output should now report missing dependencies section
+        out = runCLI(`lint ${mylib}`, { silenceError: true });
+        expect(out).toContain(
+          'Dependencies section is missing from the "package.json"'
```

Comment:
Either list out which dependencies are missing or fail the rule multiple times for each package that is missing.